docs/features/pixi_global_manifest.md

```diff
@@ -0,0 +1,222 @@
+
+## Manifest
+
+```toml title="pixi-global.toml"
+# The name of the environment is `python`
+# It will expose python, python3 and python3.11, but not pip
+[envs.python.dependencies]
+python = { spec = "3.11.*", expose_binaries = "auto" }
+pip = "*"
+
+# The name of the environment is `python_3_10`
+# It will expose python3.10
+[envs.python_3_10.dependencies]
+python = { spec = "3.10.*", expose_binaries = { "python3.10"="python" } }
+
+```
+
+## CLI
+
+Install one or more packages `PACKAGE` into their own environments and expose their binaries.
+If no environment named `PACKAGE` exists, it will be created.
+The syntax for `MAPPING` is `exposed_name=binary_name`, so for example `python3.10=python`.
+
+```
+pixi global install [--expose MAPPING] <PACKAGE>...
+```
+
+Remove environments `ENV`.
+```
+pixi global uninstall [ENV]...
+```
+
+Upgrade all packages in environments `ENV`
+```
+pixi global upgrade [ENV]...
+```
+
+Inject package `PACKAGE` into an existing environment `ENV`.
+If environment `ENV` does not exist, it will return with an error.
+```
+pixi global inject --environment ENV PACKAGE
+```
+
+Remove package `PACKAGE` from environment `ENV`.
+If that was the last package remove the whole environment and print that information in the console.
+```
+pixi global remove --environment ENV PACKAGE
+```
+
+Update the version of one package
+```
+pixi global update --environment ENV PACKAGE
+```
+
+Set for a specific package `PACKAGE` in environment `ENV` under which `MAPPING` binaries are exposed
+The syntax for `MAPPING` is `exposed_name=binary_name`, so for example `python3.10=python`.
+```
+pixi expose --environment ENV --package PACKAGE [MAPPING]...
+```
+
+Ensure that the environments on the machine reflect the state in the manifest.
+The manifest is the single source of truth.
+Only if there's no manifest, will the data from existing environments be used to create a manifest.
+`pixi global sync` is implied by most other `pixi global` commands.
+```
+pixi global sync
+```
+
+List all environments, their specs and exposed binaries
+```
+pixi global list
+```
+
+
+### Simple workflow
+
+Create environment `python`, install package `python=3.10.*` and expose all binaries of that package
+```
+pixi global install python=3.10.*
+```
+
+Upgrade all packages in environment `python`
+```
+pixi global upgrade python
+```
+
+Remove environment `python`
+```
+pixi global uninstall python
+```
+
+Create environment `python` and `pip`, install corresponding packages and expose all binaries of that packages
+```
+pixi global install python pip
+```
+
+Remove environments `python` and `pip`
+```
+pixi global uninstall python pip
+```
+
+### Injecting dependencies
+
+Create environment `python`, install package `python` and expose all binaries of that package.
+Then add package `hypercorn` to environment `python` but doesn't expose its binaries.
+
+```
+pixi global install python
+pixi global inject --environment=python hypercorn
```

Comment:
Same here with the `--environment`, this flag is required so why not just remove it? e.g.:

```suggestion
pixi global inject python hypercorn
```

Reply:
As mentioned above, I disagree. I feel like your change motivates nicely, why that isn't necessarily an improvement.
If I read `pixi global inject python hypercorn` my first thought is not that `hypercorn` will be injected in the `python` environment, but that the `python` and `hypercorn` packages will be injected somewhere.

Reply:
Also I kind of get where the confusion comes from, but otoh it does make sense that it works in that way. The first time with pipx it gave me a small pause, but after that I was instantly used to it :)

Reply:
I don't mind `--into-env` in general.

Two counterarguments:
- the name `--into-env` would be a bit weird for something like `pixi global remove/uninject`
- we also use `--environment` for pixi

Both arguments go into the direction that I expect consistency to help users.

Reply:
But can we then please have the alias `-e` and `--env` (environment is a crappy word to type)

Reply:
Why do we need `--into-env` for `remove` or `inject`? 
I think `--into-env` it's a good alternative for `--environment`

Reply:
I am even happy to only add `-e` and `--env`.
If someone objects to that, let's go with `-e`, `--env` and `--environment`.
But if we go for the latter, please let's also add the alias for the other pixi commands.

What do you say @ruben-arts?